app/Http/Controllers/BookingAPIController.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Booking;
+use App\Models\Room;
+use Carbon\CarbonPeriod;
+use http\Env\Response;
+use Illuminate\Http\Request;
+
+class BookingAPIController extends Controller
+{
+    public function create(Request $request)
+    {
+
+        $request->validate([
+            'room_id' => 'required|integer|exists:rooms,id',
+            'customer' => 'required|string|max:255',
+            'guests' => 'required|integer|min:1',
+            'start' => 'required|date|date_format:Y-m-d|after_or_equal:today',
+            'end' => 'required|date|date_format:Y-m-d|after:start',
+        ]);
+
+        $room = Room::find($request->room_id);
+        $startDate = $request->start;
+        $endDate = $request->end;
+
+
+        $bookingConflict = Booking::where('room_id', $room->id)
+            ->where(function ($query) use ($startDate, $endDate) {
+                $query->whereBetween('start', [$startDate, $endDate])
+                    ->orWhereBetween('end', [$startDate, $endDate])
+                    ->orWhere(function ($query) use ($startDate, $endDate) {
+                        $query->where('start', '<=', $startDate)
+                            ->where('end', '>=', $endDate);
+                    });
+            })
+            ->exists();
+
+        if ($room->min_capacity > $request->guests OR  $room->max_capacity < $request->guests) {
```

Comment:
You could also put this condition into the `BookingService` - that would improve readability in this file.

If you had something that looked like this instead:

```
if (BookingService::guestsAreValid($room, $request->guests) {
....
}

```

It makes it more obvious what this if statement is doing. And again, this may well be a check that you'd need to repeat in the future.